tests/annotator/annotator.cpp

```diff
@@ -562,15 +564,19 @@ TEST(Annotator, getItemBySpecificTypeDuplicateId)
     EXPECT_EQ(nullptr, annotator->importSource("duplicateId"));
 
     // Expect errors because the annotator library hasn't been built and no model exists.
-    EXPECT_EQ(size_t(30), annotator->errorCount());
+    EXPECT_EQ(size_t(15), annotator->errorCount());
+    for (size_t i = 0; i < 15; ++i) {
+        EXPECT_EQ("This Annotator object does not have a model to work with.", annotator->error(i)->description());
+    }
 
-    annotator->buildModelIndex(model);
+    annotator->setModel(model);
 
     // Expect that the errors have been cleared.
     EXPECT_EQ(size_t(0), annotator->errorCount());
 
-    EXPECT_EQ(nullptr, annotator->model("duplicateId"));
-    EXPECT_EQ(nullptr, annotator->encapsulation("duplicateId"));
+    EXPECT_NE(nullptr, annotator->model("duplicateId"));
+    EXPECT_NE(nullptr, annotator->encapsulation("duplicateId"));
+
```

Comment:
I reeeaaaaeally don't agree with this change in behaviour!  I think it's really dodgy to not alert the user that the item found has a non unique id, and just return the first one we happen to come to.  This is not great ... :(

Reply:
If the documentation of the method explains the behaviour then I think this is the correct.  The only thing I could see it doing is raising a warning that this identifier is not unique.

Reply:
I think this will end up causing all kinds of problems.  It's naive and assumes that the users know our internal index order at which their components/variables/everything is stored ... which is kinda ridiculous as we ourselves don't ever rely on it!  Being able to interact with duplicated ids is a convenience function so that users can fix their own mistakes - I think it's dangerous to just hide the fact that it's duplicated from the user.  If they want to retrieve duplicated things they should use explicit functions to do that so that everyone knows exactly what they're dealing with.

Reply:
I think so it is now back to the original setting, @kerimoyle can correct me if i'm wrong here.

Reply:
Yes, I'm happier now, thanks.